src/main/java/model/Car.java

```diff
@@ -0,0 +1,125 @@
+package model;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.Arrays.*;
+import static java.util.stream.Collectors.toList;
+
+public class Car {
+    private int score;
+
+    private final String name;
+
+    public static final int START_SCORE = 1;
+
+    public Car(String name, int score) {
+        this.name = name;
+        this.score = score;
+    }
+
+    public static List<Car> getWinner(List<Car> carList) {
+        int maxScore = getMaxScore(carList);
+
+        List<Car> maxCars = carList.stream()
+            .filter(car -> car.isMaxScoreCar(maxScore))
+            .collect(toList());
+
+        return maxCars;
+    }
+
+    public static List<String> getWinnerName(List<Car> carList) {
+        return getWinner(carList).stream()
+            .map(car -> car.name)
+            .collect(toList());
+    }
+
+    private boolean isMaxScoreCar(int maxScore) {
+        return this.score == maxScore;
+    }
+
+    private static int getMaxScore(List<Car> carList) {
+        return carList.stream()
+            .map(car -> car.score)
+            .max(Integer::compareTo)
+            .orElse(START_SCORE);
+    }
+
+    public static List<Car> getCar(String[] carNameList) {
+        return stream(carNameList)
+            .map(carName -> {
+                return new Car(carName, START_SCORE);
+            })
+            .collect(toList());
+    }
+
+    public static void validate(String str) {
+        String[] carNameList = str.split(",");
+        HashSet<String> checker = new HashSet<>();
+
+        for (String carName : carNameList) {
+            validateCarNameLength(carName);
+            checker.add(carName);
+        }
+
+        if (checker.size() != carNameList.length) {
+            throw new IllegalArgumentException("동일한 차량 이름은 불가능합니다.");
+        }
+    }
+
+    private static void validateCarNameLength(String str) {
+        if (str.length() > 5) {
+            throw new IllegalArgumentException("차량 이름은 5자를 넘어서는 안됩니다.");
+        }
+    }
+
+
+    public static Car updateCarScore(Car car, int num) {
+        if (num > 3){
+            car.score++;
+        }
+
+        return new Car(car.name, car.score);
+    }
+
+    public static List<Car> updateCarList(List<Car> initCarList, List<Integer> randomNumList) {
+        List<Car> updatedCarList = new ArrayList<>();
+
+        for (int i=0; i<initCarList.size(); i++) {
+            updatedCarList.add(updateCarScore(initCarList.get(i), randomNumList.get(i)));
+        }
```

Comment:
지금처럼 반복문을 사용하면 반복문 루프를 돌 때마다 매번 `size()` 함수를 호출하게 될거에요. 
(어떤 메서드냐에따라서 매우 비효율적인 코드가 될 수 있어요. 🥲)
아래처럼 변수로 뽑아내서 다시 호출되지 않도록 변경해주세요. 🙏

```suggestion
        int carSize = initCarList.size();
        for (int i = 0; i < carSize; i++) {
            updatedCarList.add(updateCarScore(initCarList.get(i), randomNumList.get(i)));
        }
```

Reply:
[d58ca7e](https://github.com/next-step/java-racingcar-kakao/pull/31/commits/d58ca7e0e384245f5eecb029af8530bda99c9f7b)
변수로 수정했습니다.